mk2/launchpad.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2013 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package launchpad provides interfaces to talk to
+// Novation Launchpads via MIDI in and out.
+package mk2
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/rakyll/portmidi"
+)
+
+// Launchpad represents a device with an input and output MIDI stream.
+type Launchpad struct {
```

Comment:
I was thinking whether we should put this in the top-level launchpad package and call the type MK2 instead.

```
type MK2 struct {}

func OpenMK2() (*MK2, error)
```